parser-typechecker/unison/Main.hs

```diff
@@ -122,40 +137,62 @@ main = do
           theCodebase <- FileCodebase.getCodebaseOrExit mcodepath
           let fileEvent = Input.UnisonFileChanged (Text.pack "<standard input>") contents
           launch currentDir configFilePath theCodebase [Left fileEvent, Right $ Input.ExecuteI mainName, Right Input.QuitI]
-    "transcript" : args -> runTranscripts False mcodepath args
-    "transcript.fork" : args -> runTranscripts True mcodepath args
+    "transcript" : args' -> 
+      case args' of
+      "-preserve" : transcripts -> runTranscripts False True mcodepath transcripts
+      _                         -> runTranscripts False False mcodepath args'
+    "transcript.fork" : args' -> 
+      case args' of
+      "-preserve" : transcripts -> runTranscripts True True mcodepath transcripts
+      _                         -> runTranscripts True False mcodepath args'
     _ -> do
       PT.putPrettyLn usage
       Exit.exitWith (Exit.ExitFailure 1)
 
-runTranscripts :: Bool -> Maybe FilePath -> [String] -> IO ()
-runTranscripts inFork mcodepath args = do
+prepareTranscriptDir :: Bool -> Maybe FilePath -> IO FilePath
+prepareTranscriptDir inFork mcodepath = do
+  currentDir <- getCurrentDirectory
+  tmp <- Temp.createTempDirectory currentDir "transcript"
+
+  unless inFork $ do 
+    PT.putPrettyLn . P.wrap $ "Transcript will be run on a new, empty codebase."
+    _ <- FileCodebase.initCodebase tmp
+    pure()
+
+  when inFork $ FileCodebase.getCodebaseOrExit mcodepath >> do
+    origCodePath <- FileCodebase.getCodebaseDir mcodepath
+    let path = origCodePath FP.</> FileCodebase.codebasePath
+    PT.putPrettyLn $ P.lines [
+      P.wrap "Transcript will be run on a copy of the codebase at: ", "",
+      P.indentN 2 (P.string path)
+      ]
+    Path.copyDir path (tmp FP.</> FileCodebase.codebasePath)
+
+  pure tmp
+
+cleanup :: FilePath -> IO ()
+cleanup transcriptDir = do
+  PT.putPrettyLn $ P.lines [
+    P.wrap "Deleting temporary directory: ", "",
+    P.indentN 2 (P.string transcriptDir)
+    ]
```

Comment:
Similarly, I think we can eliminate this message.

Reply:
My thinking behind originally was to make it "observable" so I could test with a script containing all flag combinations. Removing it, as it exposes underlying implementation detail. Subsequently there's no point to the wrapper, so calling `removeDirectoryRecursive` directly